### ISSUE TITLE
Fix/Remove features after new patch

### ISSUE THREAD

**Issue by ThatBigDerp** [ArmynC/ArminC-AutoExec#10]
>      Certain types of movement/shooting input automation such as hardware-assisted counter strafing will now be detected on Valve official servers, resulting in a kick from the match
>     Input binds that include more than one of the following commands will now be ignored by default. Support can be re-enabled using the cheat-protected convar `cl_allow_multi_input_binds 1`
>         sprint, reload, attack, attack2, turnleft, turnright, turnup, turndown, forward, back, left, right, moveup, movedown, klook, use, jump, duck, strafe, zoom, yaw, pitch, forwardback, rightleft
>     The jump-throw confirmation grunt sound can now be heard by other players nearby

After the new update some of the binds no longer function like jumpthrow, runthrow and some more.

**Comment by ThatBigDerp** [comment]
Update: At your own risk here is fixed jumpthrow

```
alias revert "bind mouse_x yaw"
alias combo "-jump;revert"
alias +jumpthrow "+jump"
alias -jumpthrow "-attack; bind mouse_x combo"

bind x +jumpthrow
```
 props to poggu__ on X for this

**Comment by armync** [comment]
Thanks for letting me know! That's unfortunate. I'll wait a bit for their next reaction... Maybe they'll retract the update due to community pressure.

Interesting finding with that bind. Though, I will not implement it. Yet. There's is a certainty of Valve blocking that too and the autoexec's users getting kicked without even know due to what. If Valve doesn't revert to the initial manner and nothing happens because of this script, I will implement it in no time.

ref (in case): https://x.com/poggu__/status/1825663353336811747

--
![image](https://github.com/user-attachments/assets/8c1e19a5-3e0f-4fe3-89f2-954df0d3ae9e)

Hope not :joy:

**Comment by armync** [comment]
Update:

* AFK Move is broken.
* Alternate binds are broken; the entire keyboard binding will mess up by pressing ALT.

**Comment by ThatBigDerp** [comment]
Also found this gem on reddit:
```
-forward
-back
-right
-left
rightleft 0 0 0
forwardback 0 0 0
alias"+a""rightleft -1 0 0;alias +d +ad"
alias"a1""rightleft -1 0 0;alias +d +ad"
alias"+ad""rightleft 0.0001 0 0;alias -d add;alias -a ada"
alias"-ad""rightleft -0.0001 0 0;alias -d d2;alias -a a2"
alias"-a""rightleft 0.0001 0 0;alias +d d1;alias -d d2;alias -a a2;alias +a a1"
alias"a2""rightleft 0.0001 0 0;alias +d d1;alias -d d2;alias -a a2;alias +a a1"
alias"+d""rightleft 1 0 0;alias +a +da"
alias"d1""rightleft 1 0 0;alias +a +da"
alias"+da""rightleft -0.0001 0 0;alias -a ada;alias -d add"
alias"-da""rightleft 0.0001 0 0;alias -a a2;alias -d d2"
alias"-d""rightleft -0.0001 0 0;alias +a a1;alias -a a2;alias -d d2;alias +d d1"
alias"d2""rightleft -0.0001 0 0;alias +a a1;alias -a a2;alias -d d2;alias +d d1"
alias"add""rightleft -1 0 0;alias +d +ad;alias -a a2;alias -d d2"
alias"ada""rightleft 1 0 0;alias +a +da;alias -d d2;alias -a a2"
bind a +a
bind d +d
bind w +forward
bind s +back```

Auto-stop binds

**Comment by armync** [comment]
I saw new attempts at scripts, including NULL movement. ref: https://www.youtube.com/watch?v=Q9cTcKg-nVM

On this subject, **I wouldn't endorse their usage at ALL**. Due to the current changes and even older/present tournaments rules, using these may and will result in penalties and even ban (talking about professional tournaments).

![image](https://github.com/user-attachments/assets/5d61fe8b-b561-4be6-abc2-46be033bc842)

ref: [ESL Pro Tour - CS Game Specific Rules 2024.pdf](https://github.com/user-attachments/files/16691267/ESL.Pro.Tour.-.CS.Game.Specific.Rules.2024.pdf)

On the other hand, an simple and fair usage of autoexec would be clear under these conditions:

![image](https://github.com/user-attachments/assets/871ac10d-16ee-436a-86cc-2b844863e280)

This autoexec is aiming to be functional in most case-scenarios, be it from simple to professional gaming. As such, **I can't implement** NULL scripts in official releases. However, documenting them in discussions is an interesting approach. This way, anyone who wants to use them can easily copy the code.

--

I've released a new autoexec version, hotfix 00d74f6. Now, everything should work fine, including AFK Movement script, Second Binds and so on. The Jump Throw script was deprecated (for now). I'll check the current state of rules and Valve decisions. Now, the jump throws are easily doable. Also, on this matter, is also a problem based on fair approach of gaming skill. Should one use bypass-scripts that can enable jump-throw even if the rest of players aren't using them?...

**Comment by armync** [comment]
Added the missing manual jump-throw bind with 8d32e56. This ensures that even if the throw system is 'forgiving' on mistakes, now it should be very smooth.

The usage is very simple. Place the thumb/index finger on both jump (_space_) and manual jump-throw bind (currently _n_), then press them at the same time. This will result in a synchronized jump throw.
